### ISSUE TITLE
Invalid attestations in the slashing protection export

### ISSUE THREAD

**Issue by realbigsean** [prysmaticlabs/prysm#8893]
# 🐞 Bug Report

### Description

We had a user trying to migrate their slashing protection database from prysm to lighthouse.  Their JSON export had some entries that would have been slashable but were unsigned, so Lighthouse was unable to import them:

<img width="413" alt="Screen Shot 2021-05-14 at 1 53 12 PM" src="https://user-images.githubusercontent.com/5160426/118310115-2c98e100-b4bc-11eb-82ee-49878b311b0f.png">

## 🌍  Your Environment

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
1.3.9
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

**Comment by michaelsproul** [comment]
I'm trying to understand the root cause of this bug, but I'm having trouble.

I'm guessing that the `0 --> 10488` attestation without a signing root is some sort of synthetic attestation that gets inserted into the DB in special circumstances? However I can't find the spot in your code where you do this.

I thought that maybe the synthetic attestation got created when [importing an interchange file](https://github.com/prysmaticlabs/prysm/blob/f2fe1f7683fb80272bd168e6a4a0a10be878e604/validator/slashing-protection/local/standard-protection-format/import.go#L25), in order to update the low watermark for source/target. A synthetic attestation like `0 --> n` would prevent new messages being signed with `0 < source < target` and `target <= n`, which _kind of_ satisfies the low watermark conditions with some issues. E.g. it's still possible to sign `0 --> n - 1` (it's not surrounded).

I noticed in the JSON data that the attesters seem to have a gap in their signing history around the time of the synthetic attestation. Specifically, the synthetic attestation is always `0 --> n` and the first attestation after the gap is `n + 1 --> n + 2`. If the intention of the synthetic attestation is to set the watermarks after importing history, then there are other cases besides `0 --> n - 1` where it will fail to do this. E.g. it would be possible to sign an attestation `n --> n + 1` which violates the `source >= new_min_source` requirement (noting `new_min_source = n + 1`).

Lighthouse used to do something similar, but I changed it after we had trouble updating the watermark when importing a 2nd interchange file. I modified the standard tests to try to cover these cases, and updated the advice in the [Security Considerations](https://eips.ethereum.org/EIPS/eip-3076#security-considerations) section of EIP-3076:

> Care must be taken to avoid signing messages within a gap in the database (an area of unknown signing activity). This could occur if two interchanges were imported with a large gap between the last entry of the first and the first entry of the second. Signing in this gap is not safe, and would violate conditions (2), (4) and (5). It can be avoided by storing an explicit low watermark in addition to the actual messages of the slashing protection database, or by pruning on import so that the oldest messages from the interchange become the oldest messages in the database.

My other guess was that maybe the synth gets inserted on start-up, or was inserted by some previous version of Prysm? But I think I need your help to track that down.

I'd like to get to the root cause so we can update the standard tests to cover this case, and the text of the EIP if necessary.

Thanks!
